pkg/kubectl/cmd/explain.go

```diff
@@ -76,9 +76,10 @@ func NewCmdExplain(parent string, f cmdutil.Factory, streams genericclioptions.I
 	cmd := &cobra.Command{
 		Use: "explain RESOURCE",
 		DisableFlagsInUseLine: true,
-		Short:   i18n.T("Documentation of resources"),
-		Long:    explainLong + "\n\n" + cmdutil.SuggestApiResources(parent),
-		Example: explainExamples,
+		Short:     i18n.T("Documentation of resources"),
+		Long:      explainLong + "\n\n" + cmdutil.SuggestApiResources(parent),
+		Example:   explainExamples,
+		ValidArgs: cmdutil.AllResourceList(f),
```

Comment:
this contacting the server during command construction, which isn't appropriate (it means this runs every time you invoke kubectl, regardless of whether you are running `explain`, and if a server is unavailable, can hang running any command or even displaying help)

Reply:
opened https://github.com/kubernetes/kubernetes/pull/63421 as an alternative that does the lookup when needed for completion